gtsfm/utils/metrics.py

```diff
@@ -256,3 +242,119 @@ def log_sfm_summary() -> None:
         "Averaging median_trans_dist_err: %.2f", averaging_metrics["translation_averaging_distance"]["median_error"]
     )
     logger.info("Averaging max_trans_dist_err: %.2f", averaging_metrics["translation_averaging_distance"]["max_error"])
+
+
+def persist_frontend_metrics_full(
+    two_view_report_dict: Dict[Tuple[int, int], TwoViewEstimationReport], images: List[Image]
+) -> None:
+    """Persist the front-end metrics for every pair on disk.
```

Comment:
should we add a comment about why we arent using GtsfmMetric for this one?

I guess because we want to debug individual image pairs, rather than just looking at the summary?

Reply:
Thanks for all the updates Akshay. Looks good. Bump on this one.

Reply:
GtsfmMetric does not support the representation of this json, where each value is sort of a dict of other values. 
Is that what we should document? I did not feel that is necessary because theres no question of using GtsfmMetric in the first place, but I can add it if you feel its not obvious.